hooks/useAssetMetadata.ts

```diff
@@ -0,0 +1,48 @@
+import { BridgedAssetsMetadata } from "@/lib/snowbridge";
+import {
+  assetErc20MetaDataAtom as assetErc20MetadataAtom,
+  relayChainNativeAssetAtom,
+  snowbridgeEnvironmentAtom,
+} from "@/store/snowbridge";
+import { useAtom, useAtomValue } from "jotai";
+import { useEffect } from "react";
+import useSWR from "swr";
```

Comment:
SWR stands for stale-while-revalidate. Its is a pattern for fetching data in which "stale" cached data is always returned so that the UI is rendered immediately, the data can be fetched in the background and updated once the new data has arrived.

In this example we are using it for asset meta data, which wont change often is probably ok if the user sees stale data. At worst a new asset added will pop into view a few seconds later.

We use this npm package to automate.
https://swr.vercel.app/